CoreScriptsRoot/Modules/Settings/Pages/ReportAbuseMenu.lua

```diff
@@ -165,6 +165,45 @@ local function Initialize()
 			submitText.ZIndex = 1
 		end
 
+		local function areFieldsCompleted()
+			-- Returns: Bool completed
+			-- Tells you if the fields required for submission have been filled out
+
+			local completed = false
+
+			local gameOrPlayerIndex = this.GameOrPlayerMode.CurrentIndex
+			local selectedPlayerIndex = this.WhichPlayerMode:GetSelectedIndex()
+			local selectedAbuseTypeIndex = this.TypeOfAbuseMode:GetSelectedIndex()
+			
+			if gameOrPlayerIndex == 1 then -- Game type abuse
+				local abuseTypeIsSelected = selectedAbuseTypeIndex ~= nil and selectedAbuseTypeIndex > 0
+
+				completed = abuseTypeIsSelected
+			elseif gameOrPlayerIndex == 2 then -- Player type abuse
+				local playerIsSelected = selectedPlayerIndex ~= nil and selectedPlayerIndex > 0
+				local abuseTypeIsSelected = selectedAbuseTypeIndex ~= nil and selectedAbuseTypeIndex > 0
+
+				completed = playerIsSelected and abuseTypeIsSelected
+			end
+
+			return completed
+		end
+
+		local function updateSubmitButtonState()
+			-- Returns: Bool active
+			-- Activates/deactivates the submit button based on current state
+
+			local active = areFieldsCompleted()
```

Comment:
a little weird to have active be the name from areFieldsCompleted... maybe use fieldsCompleted?